tests/common/mod.rs

```diff
@@ -0,0 +1,228 @@
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        match msg {
+            SeiMsg::CreateDenom { subdenom } => {
+                let denom = format!("factory/{}/{}", sender, subdenom);
+                if storage.get(denom.as_bytes()).is_some() {
+                    return Err(anyhow::anyhow!("denom already exists"));
+                }
+
+                storage.set(denom.as_bytes(), sender.to_string().as_bytes());
+
+                Ok(AppResponse {
+                    events: vec![],
+                    data: Some(to_binary(&denom).unwrap()),
+                })
+            }
+            SeiMsg::MintTokens { amount } => {
+                let owner = storage.get(amount.denom.as_bytes());
+                if owner.is_none() || owner.unwrap() != sender.to_string().as_bytes() {
+                    return Err(anyhow::anyhow!(
+                        "Must be owner of coin factory denom to mint"
+                    ));
+                }
+                router.sudo(
+                    api,
+                    storage,
+                    block,
+                    SudoMsg::Bank(BankSudo::Mint {
+                        to_address: sender.to_string(),
+                        amount: vec![amount],
+                    }),
+                )
+            }
+            SeiMsg::BurnTokens { amount } => {
+                let owner = storage.get(amount.denom.as_bytes());
+                if owner.is_none() || owner.unwrap() != sender.to_string().as_bytes() {
+                    return Err(anyhow::anyhow!(
+                        "Must be owner of coin factory denom to burn"
+                    ));
+                }
+                Ok(router
+                    .execute(
+                        api,
+                        storage,
+                        block,
+                        sender,
+                        CosmosMsg::Bank(BankMsg::Burn {
+                            amount: vec![amount],
+                        }),
+                    )
+                    .unwrap())
+            }
+            _ => panic!("Unexpected custom exec msg"),
+        }
+    }
+
+    fn query(
+        &self,
+        _api: &dyn Api,
+        _storage: &dyn Storage,
+        _querier: &dyn Querier,
+        _block: &BlockInfo,
+        request: Self::QueryT,
+    ) -> AnyResult<Binary> {
+        match request.query_data {
+            SeiQuery::ExchangeRates {} => Ok(Binary::default()),
+            SeiQuery::OracleTwaps { lookback_seconds } => Ok(Binary::default()),
+            SeiQuery::DexTwaps {
+                contract_address,
+                lookback_seconds,
+            } => Ok(Binary::default()),
+            SeiQuery::OrderSimulation {
+                order,
+                contract_address,
+            } => Ok(Binary::default()),
+            SeiQuery::Epoch {} => Ok(to_binary(&EpochResponse {
```

Comment:
for epoch, can we actually use the app stored data instead, so we can have epochs that are more realistic in behavior rather than constant? its fine if we leave as a TODO here and leave constant for now, but I don't want it to be forgotten because it looks like its implemented already.